src/screencast/pipewire_screencast.c

```diff
@@ -72,6 +72,9 @@ static void pwr_handle_stream_state_changed(void *data,
 	switch (state) {
 	case PW_STREAM_STATE_STREAMING:
 		cast->pwr_stream_state = true;
+		if (!cast->wlr_frame) {
+			xdpw_wlr_register_cb(cast);
+		}
```

Comment:
Checked every commit again and found a race condition only in `screencast: only restart wlroots loop if stream is active` 
when PipeWire sends a `state_changed` changing the state to `PW_STATE_STREAMING` in between the `buffer_done` and the `ready`/`failed` events  of the screencopy loop. This did trigger a new screencopy loop entrance while the old wasn't finished and triggered an assertion in `xdpw_pwr_dequeue_buffer`.

Checking if **NO** screencopy loop is currently running via `cast->wlr_frame == NULL` solved this issue.